src/com/rath/rathbot/cmd/disc/actions/BanCmd.java

```diff
@@ -56,51 +59,29 @@ public boolean executeCommand(final IMessage msg, final String[] tokens, final i
       return RBCommand.STOP_CMD_SEARCH;
     }
     
-    // Tests if the first argument of the command is an @mention or uid, then processes the argument accordingly.
-    long banUserID = 0;
-    final String userToken = tokens[tokDepth + 1];
-    if (userToken.matches("<@!?\\d+>")) {
-      
-      int hasNickName = 0;
-      
-      if (userToken.charAt(2) == '!') hasNickName = 1;
-      
-      // If argument is @mention, substring to get UID
-      try {
-        banUserID = Long.parseLong(
-            userToken.substring((userToken.indexOf('@') + hasNickName + 1), userToken.indexOf('>')));
-      } catch (NumberFormatException nfe) {
-        nfe.printStackTrace();
-      }
-    } else {
-      
-      // Else argument is UID, parse UID
-      try {
-        banUserID = Long.parseLong(tokens[tokDepth + 1]);
-      } catch (@SuppressWarnings("unused") NumberFormatException nfe) {
-        RathBot.sendMessage(channel, "Invalid UID. UIDs must only contain numbers.");
-      }
-    }
-    
-    final IDiscordClient client = RathBot.getClient();
-    
-    // Create IUser object from banUserID to ban them, also verify valid UID.
-    final IUser bannedUser = client.getUserByID(banUserID);
-    if (bannedUser == null) {
-      RathBot.sendMessage(channel, "Error! User not found, please enter a valid UID.");
+    // Create IUser object from token to issue disciplinary action on them.
+    final IUser infringingUser = MessageHelper.getUserFromToken(tokens[tokDepth + 1], channel);
+    if (infringingUser == null) {
+      RathBot.sendMessage(channel,
```

Comment:
Eclipse will auto-format long strings more nicely when you break them up and concatenate them.

Do this:
```
System.out.println("This is a really long String. It's so long that I'm " + 
                    "running out of room!");
```

Not this:
```
System.out.println(
         "Eclipse doesn't like to automatically break apart strings, so it does this, and it looks ugly.");
```